pkg/cloud/scope/machine.go

```diff
@@ -181,6 +182,42 @@ func (m *MachineScope) SetAnnotation(key, value string) {
 	m.AWSMachine.Annotations[key] = value
 }
 
+// SyncMachineNodeLabel sets a key value label on the AWSMachine node.
+func (m *MachineScope) SyncMachineNodeLabel(ctx context.Context, remoteClient client.Client) error {
```

Comment:
Is `SyncMachineNodeLabel` really accurate as a function name?

Looks like we only ever copy labels if they have the correct prefix. There's no ability to remove these labels.